frontend/src/layout/header/UserBox.tsx

```diff
@@ -33,71 +33,135 @@ export const UserBox: React.FC = () => {
         fontSize: 22,
         opacity: 0.4,
     };
-    if (user === "unknown") {
-        return <Spinner css={iconCss} />;
-    } else if (user === "error") {
-        // TODO: tooltip
-        return <FiAlertTriangle css={iconCss} />;
-    } else if (user === "none") {
-        return <LoggedOut />;
-    } else {
-        return <LoggedIn {...{ t, user }} />;
-    }
+
+    return <>
+        <LanguageSettings />
+        {user === "unknown"
+            ? <Spinner css={iconCss} />
+            : user === "error"
+                // TODO: tooltip
+                ? <FiAlertTriangle css={iconCss} />
+                : user === "none"
+                    ? <LoggedOut />
+                    : <LoggedIn {...{ t, user }} />
+        }
+    </>;
 };
 
 
 /** User-related UI in header when the user is NOT logged in. */
 const LoggedOut: React.FC = () => {
     const { t } = useTranslation();
 
-    return (
-        <div css={{ display: "flex", padding: "8px 0" }}>
-            <Link
-                to={CONFIG.auth.loginLink ?? LOGIN_PATH}
-                onClick={() => {
-                    // Store a redirect link in session storage.
-                    window.sessionStorage.setItem(REDIRECT_STORAGE_KEY, window.location.href);
-                }}
-                htmlLink={!!CONFIG.auth.loginLink}
-                css={{
-                    alignSelf: "center",
-                    borderRadius: 10,
-                    cursor: "pointer",
-                    padding: "5px 14px",
-                    marginRight: 8,
-                    display: "flex",
-                    gap: 8,
-                    alignItems: "center",
-                    backgroundColor: "var(--nav-color)",
-                    color: "var(--nav-color-bw-contrast)",
-                    "&:hover, &:focus": {
+    return <>
+        <Link
+            to={CONFIG.auth.loginLink ?? LOGIN_PATH}
+            onClick={() => {
+                // Store a redirect link in session storage.
+                window.sessionStorage.setItem(REDIRECT_STORAGE_KEY, window.location.href);
+            }}
+            htmlLink={!!CONFIG.auth.loginLink}
+            css={{
+                outline: "transparent",
+                ":hover, :focus": {
+                    "> :first-child": {
                         backgroundColor: "var(--nav-color-dark)",
                         color: "var(--nav-color-bw-contrast)",
                     },
```

Comment:
Why not move that into the `css` prop of that first child below?

Reply:
That would work for hover, but not for focus, as the parent element is the one being focused (if I understand things correctly). I could split this up, but I don't really think that makes it more readable.

On the plus side, I realized that I can pull the duplicate `> div: opacity: 1` from the selectors below into this expression, and change the `> div:` to `> :last-child:` as the opacity is only needed on the icon only version of the button (not that this is necessary, but at least it's more accurate).